commands/deploy.go

```diff
@@ -249,7 +250,34 @@ Error: %s`, fprocessErr.Error())
 				function.ReadOnlyRootFilesystem = deployFlags.readOnlyRootFilesystem
 			}
 
-			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, annotations, functionResourceRequest1, function.ReadOnlyRootFilesystem, tlsInsecure)
+			// Need to alter Gateway to allow nil/empty string as fprocess, to avoid this repetition.
+			var fprocessTemplate string
+			if len(function.FProcess) > 0 {
+				fprocessTemplate = function.FProcess
+			}
+			req := requests.CreateFunctionRequest{
+				EnvProcess:             fprocessTemplate,
+				Image:                  function.Image,
+				RegistryAuth:           function.RegistryAuth,
+				Network:                services.Provider.Network,
+				Service:                function.Name,
+				EnvVars:                allEnvironment,
+				Constraints:            functionConstraints,
+				Secrets:                deployFlags.secrets,
+				Labels:                 &allLabels,
+				Annotations:            &annotations,
+				ReadOnlyRootFilesystem: function.ReadOnlyRootFilesystem,
+			}
+
+			statusCode := proxy.DeployFunction(
+				req,
+				services.Provider.GatewayURL,
+				function.Language,
+				deployFlags.replace,
+				deployFlags.update,
+				functionResourceRequest1,
```

Comment:
do we know why this variable is called `functionResourceRequest1`?  I don't see any other `functionResourceRequest` variables, so why the suffix? If we are renaming variables, perhaps this one could lose the suffix

Reply:
My goal was to not modify the structs, but to only add my additional field. I would be more then happy to modify this field, as I completely agree the suffix is unnecessary.